src/routes/courierPage.tsx

```diff
@@ -1,314 +1,314 @@
-import * as React from "react";
-import { createFileRoute, Link } from "@tanstack/react-router";
-import { useState, useEffect } from "react";
-import {
-  getFirestore,
-  collection,
-  addDoc,
-  doc,
-  getDocs,
-  query,
-  where,
-  updateDoc,
-  Timestamp,
-} from "firebase/firestore";
-import Select, { SingleValue } from "react-select";
-import Swal from "sweetalert2";
-
-export const Route = createFileRoute("/courierPage")({
-  component: RouteComponent,
-});
-
-type UserOption = {
-  value: string;
-  label: string;
-};
-
-type CompartmentOption = {
-  id: string;
-  compartment: string | number; // Updated to handle both string and number
-};
-
-function RouteComponent() {
-  const [users, setUsers] = useState<UserOption[]>([]);
-  const [selectedUser, setSelectedUser] =
-    useState<SingleValue<UserOption>>(null);
-  const [compartmentSizes] = useState([
-    { value: "Extra Large", label: "Extra Large" },
-    { value: "Large", label: "Large" },
-    { value: "Medium", label: "Medium" },
-    { value: "Small", label: "Small" },
-  ]);
-  const [selectedSize, setSelectedSize] =
-    useState<SingleValue<{ value: string; label: string }>>(null);
-  const [availableCompartments, setAvailableCompartments] = useState<
-    CompartmentOption[]
-  >([]);
-  const [selectedCompartment, setSelectedCompartment] = useState<string>("");
-  const [selectedCourier, setSelectedCourier] = useState<string>("Amazon");
```

Comment:
Take a look into `useReducer` hook to simplify this https://react.dev/reference/react/useReducer